neurolib/optimize/exploration/exploration.py

```diff
@@ -249,44 +269,83 @@ def run(self, **kwargs):
         """
 
         self.loadDfResults(filename, trajectoryName)
 
         # make a list of dictionaries with results
-        logging.info("Creating results dictionary ...")
-        self.results = []
-        for rInd in tqdm.tqdm(range(self.nResults), total=self.nResults):
-            self.pypetTrajectory.results[rInd].f_load()
-            result = self.pypetTrajectory.results[rInd].f_to_dict(fast_access=True, short_names=pypetShortNames)
-            result = dotdict(result)
-            self.pypetTrajectory.results[rInd].f_remove()
-            self.results.append(result)
-
-        # Postprocess result keys if pypet short names aren't used
-        # Before: results.run_00000001.outputs.rates_inh
-        # After: outputs.rates_inh
-        if pypetShortNames == False:
-            for i, r in enumerate(self.results):
-                new_dict = dotdict({})
-                for key, value in r.items():
-                    new_key = "".join(key.split(".", 2)[2:])
-                    new_dict[new_key] = r[key]
-                self.results[i] = copy.deepcopy(new_dict)
+        self.results = dotdict({})
+        if all:
+            logging.info("Loading all results to `results` dictionary ...")
+            for rInd in tqdm.tqdm(range(self.nResults), total=self.nResults):
+
+                # check if enough memory is available
+                if memory_cap:
+                    assert isinstance(memory_cap, (int, float)), "`memory_cap` must be float."
+                    assert (memory_cap > 0) and (memory_cap < 100), "`memory_cap` must be between 0 and 100"
+                    # check ram usage with psutil
+                    used_memory_percent = psutil.virtual_memory()[2]
+                    if used_memory_percent > memory_cap:
+                        raise MemoryError(f"Memory use is at {used_memory_percent}% and capped at {memory_cap}. Aborting.")
+                        
+                self.pypetTrajectory.results[rInd].f_load()
+                result = self.pypetTrajectory.results[rInd].f_to_dict(fast_access=True, short_names=pypetShortNames)
+                result = dotdict(result)
+                self.pypetTrajectory.results[rInd].f_remove()
+                self.results[rInd] = copy.deepcopy(result)
+
+            # Postprocess result keys if pypet short names aren't used
+            # Before: results.run_00000001.outputs.rates_inh
+            # After: outputs.rates_inh
+            if not pypetShortNames:
+                for i, r in self.results.items():
+                    new_dict = dotdict({})
+                    for key, value in r.items():
+                        new_key = "".join(key.split(".", 2)[2:])
+                        new_dict[new_key] = r[key]
+                    self.results[i] = copy.deepcopy(new_dict)
+
+        self.aggregateResultsToDfResults()
 
         logging.info("All results loaded.")
 
+    def aggregateResultsToDfResults(self):
```

Comment:
New function to aggregate all simulation results to the pandas dataframe. Can be used without loading all results to RAM but loading (and unloading) them one by one using `getRun()`